docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,181 @@
+
+Below is an example of how the `Enqueued_Scripts_Size_Check` uses shared preparations.
+
+```php
+	/**
+	 * Returns an array of shared preparations for the check.
+	 *
+	 * @return array Returns a map of $class_name => $constructor_args pairs. If the class does not
+	 *               need any constructor arguments, it would just be an empty array.
+	 */
+	public function get_shared_preparations() {
+		$demo_posts = array_map(
+			function( $post_type ) {
+				return array(
+					'post_title'   => "Demo {$post_type} post",
+					'post_content' => 'Test content',
+					'post_type'    => $post_type,
+					'post_status'  => 'publish',
+				);
+			},
+			$this->viewable_post_types
+		);
+
+		return array(
+			Demo_Posts_Creation_Preparation::class => array( $demo_posts ),
+		);
+	}
+```
+
+## Check run method
+
+The checks `run()` method will hold all the logic to test the plugin and raise any warnings or errors that are found.
+
+The run method accepts an instance of the `Check_Results` class which is used to add messages to the results list.
+
+Warnings and errors are added via the `add_message()` method which accepts 3 parameters.
+
+- `$error (bool)` - Whether the message is an error or warning. True for error, false for warning.
+- `$message (string)` - The error/warning message.
+- `$args (array)` - Additional message arguements to add context.
+  - `$code (string)` - Violation code according to the message. Default empty string.
+  - `$file (string)` - The file in which the message occurred. Default empty string (unknown file).
+  - `$line (int)` - The line on which the message occurred. Default 0 (unknown line).
+  - `$column (int)` - The column on which the message occurred. Default 0 (unknown column).
+
+In addition to adding messages, the `Check_Result` instance also provides access to the `Plugin_Context`. The plugin context is useful for getting the plugins path and urls when performing checks.
+
+Below is an example showing how to access the plugin context and add messages using the `Check_Results` instance.
+
+```php
+/**
+	 * Runs the check on the plugin and amends results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check results to amend and the plugin context.
+	 */
+	public function run( Check_Result $result ) {
+
+    // Get the absolute file path for a specific file in the plugin.
+    $plugin_file = $result->plugin()->path( 'plugin-file.php' );
+    
+    // Run logic to test the plugin for warnings/errors...
+
+    // When an issue is found add a warning.
+    $result->add_message(
+      false, 
+			'Warning message content.', 
+			array(
+				'code'   => 'warning_code',
+				'file'   => $pluging_file,
+				'line'   => 1,
+				'column' => 10,
+			)
+    );
+  }
+```
+
+## Add the Check to the Plugin Checker
```

Comment:
See my feedback on the static check documentation for the similar section. We should make sure those sections are aligned, they can have the exact same content.